Today Screen Layout

Catachthonius opened the pull request:
Added today screen to the project.

Catachthonius committed:
Added cronstraint layout and recyclerview to fragment_today.xml

Catachthonius committed:
created today_item.xml

Catachthonius committed:
Added linear layout to today_item.xml

Catachthonius committed:
Added first element of the linear layout to today_item.xml

Catachthonius committed:
Added second element of the linear layout to today_item.xml, and imported vector assets for today screen

Catachthonius committed:
Added third element of the linear layout to today_item.xml, and fixed font for first element.

Catachthonius committed:
Added fourth element of the linear layout to today_item.xml

Catachthonius committed:
Added fifth element of the linear layout to today_item.xml, and fixed padding for fourth element

Catachthonius committed:
Added sixth element of the linear layout to today_item.xml

Catachthonius committed:
Added CardView outside of the linear layout to today_item.xml. It has a constraint layout which contains 12 TextViews and 2 Views which act as separators.

Catachthonius committed:
Fixed recyclerview in fragment_today.xml

Catachthonius committed:
Created today_title.xml and added 4 TextViews to the screen

Catachthonius committed:
Added RecyclerView for the title in fragment_today.xml

Catachthonius committed:
Fixed Linear Layout in today_item.xml

Catachthonius committed:
Moved  today_title.xml content to fragment_today.xml and deleted today_title.xml

Catachthonius committed:
Improved today_item.xml by working on elements; removed linear layout and transformed the main array of elements into a constraint; fixed cardview.

Catachthonius committed:
Created new "today" package and moved TodayFragment.kt inside in order to start working on the code and make everything less chaotic.

Catachthonius committed:
Created TodayData dataclass in today package

Catachthonius committed:
Created TodayAdapter

Catachthonius committed:
Filled TodayData data class and started work on the Adapter.

Catachthonius committed:
Finished work on TodayAdapter.

Catachthonius committed:
Added onViewCreated to TodayFragment.kt.

Catachthonius committed:
Fixed Indentation

Catachthonius committed:
Extracted string resource for tvTodayDate inside fragment_today.xml

Catachthonius committed:
Changed paddingHorizontal inside RecyclerView widget for fragment_today.xml

Catachthonius committed:
Fixed Recyclerview not showing on the app; added
fixed elements of the recyclerview opening unexpectedly multiple times within the view;
fixing graphic issue with the bottom of the recyclerview.

Catachthonius committed:
Merge branch 'develop' into todayscreen

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayAdapter.kt:
```diff
@@ -0,0 +1,69 @@
+    class TodayViewHolder(val binding: TodayItemBinding) :
+        RecyclerView.ViewHolder(binding.root)
+
+    val openElementIndex = mutableListOf<Int>()
```
a cosa serve openElementIndex?

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayAdapter.kt:
```diff
@@ -0,0 +1,69 @@
+
+class TodayAdapter(val todayscreen: List<TodayData>) :
+    RecyclerView.Adapter<TodayAdapter.TodayViewHolder>() {
+    class TodayViewHolder(val binding: TodayItemBinding) :
```
Hai creato la classe ma non ci fai nulla dentro

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayAdapter.kt:
```diff
@@ -0,0 +1,69 @@
+        val model = todayscreen[position]
+
+        //recyclerveiw item elements
+        holder.binding.ora.text = model.ora
```
Tutte le operazioni relative alle view vanno eseguite all'interno del viewholder di riferimento

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayData.kt:
```diff
@@ -0,0 +1,28 @@
+data class TodayData(
+    //recyclerview item elements
+
+val ora: String,
```
i nomi dei valori, per convenzione, sono sempre in inglese e con la prima lettera in minuscolo

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayFragment.kt:
```diff
@@ -0,0 +1,289 @@
+
+        binding.todayscreen.adapter = TodayAdapter(
+            listOf(
+                TodayData(
```
da dove arrivano i dati? Dovresti avere una classe repository o più semplicemente una classe object Data, da cui leggi i dati relativi al meteo che poi usi per generare  la lista (che rappresenta la ui) da dare in pasto all'adapter

vincenzocostagliola commented on app/src/main/java/co/develhope/meteoapp/today/TodayFragment.kt:
```diff
@@ -0,0 +1,289 @@
+                TodayData(
+                    "11:00", R.drawable.sun, "31°",
+                    R.drawable.raindrop, "0%",
+                    R.drawable.arrow,
```
la lables come gli id dell erisorse non si passano mai come argomenti di una classe.
Le lables le puoi impostare nella ui prendendo la risorsa dal file delle stringhe, le immagini/icone devi rappresentarle con un altro tipo di dato (es un enum? ) che poi userai per capire quale immagine prendere.

vincenzocostagliola commented on app/src/main/res/layout/fragment_today.xml:
```diff
@@ -1,13 +1,74 @@
+    tools:context=".today.TodayFragment">
 
     <TextView
+        android:id="@+id/luogo"
```
gli id sono sempre in inglese con la prima lettere minuscola

vincenzocostagliola commented on app/src/main/res/layout/fragment_today.xml:
```diff
@@ -1,13 +1,74 @@
+        android:layout_height="wrap_content"
+        android:layout_marginTop="61dp"
+        android:fontFamily="@font/inter"
+        android:paddingStart="40dp"
```
le dimensioni non devono mai essere hardcoded ma prese dal file delle dimensioni

vincenzocostagliola commented on app/src/main/res/layout/fragment_today.xml:
```diff
@@ -1,13 +1,74 @@
+        android:paddingStart="40dp"
+        android:text="@string/today_title_location"
+        android:textColor="@color/todaytext1"
+        android:textSize="15sp"
```
vale anche per le dimensioni dei testi

vincenzocostagliola commented on app/src/main/res/layout/fragment_today.xml:
```diff
@@ -1,13 +1,74 @@
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/oggi"
```
le textview che vengono usate per il titolo nella schermata non sono fisse ma vanno gestite nella recylerview e quindi nell'adapter

Catachthonius committed:
Refactored all values inside TodayData data class

Catachthonius committed:
Created fragment_today_title.xml

Catachthonius committed:
Removed title elements from fragment_today.xml and added them to fragment_today_title.xml

Catachthonius committed:
Created data package

Catachthonius committed:
Created dimen file with dimensions for other xml files

Catachthonius committed:
Changed hardcoded dimensions in reference to dimen.xml

Catachthonius committed:
Renamed elements of xml to English and fixed TodayAdapter appropriately

Catachthonius committed:
Fixed binding for todayRecycler inside TodayFragment

Catachthonius committed:
Fixed typos in humidity-related elements

Catachthonius committed:
Created TodayViewHolder.kt and moved ViewHolder elements into it

Catachthonius committed:
Continued work on the project: separated ViewHolder into its own class; created TodayTitleViewHolder and added elements of the title to the TodayData data class; moved the list to Data object, fixed errors on fragment_today_title.xml, fixed errors in the Adapter.

Catachthonius committed:
Fixed mistake with sealed class name

Catachthonius committed:
Imported last icon from Figma

Catachthonius committed:
Created WeatherIcon enum class and uploaded Error icon

Catachthonius committed:
Continued work on the Today Fragment screen and all its elements:
-Created enum class for the icons;
-Created two separate data classes for the forecast elements;
-Continued work on the Data object, removed resources passed as direct arguments of the class;
-Created methods to assign id to image elements;
-Continued work on the TodayFragment, on  TodayViewHolder and TodayTitleViewHolder.

Catachthonius committed:
Applied created appropriate packages and moved appropriate elements inside of them, following app architecture

Catachthonius committed:
Fixed issue with TodayForecast list

Catachthonius committed:
Deleted redundant data class

Catachthonius committed:
Fixed indentation

Catachthonius committed:
Fixed bug that only showed the first icon on the recyclerview by implementing forecastIndex inside TodayForecast and in the TodayViewHolder

Catachthonius committed:
Changed import java.time.OffsetDateTime to threeten for OffsetDateTime
